:sparkles: Added new articles pages section

fenix2222 opened the pull request:
- Affected routes: /articles

- Fixed #2763 


### Screenshots

![2024-07-08_10-22-32](https://github.com/SSWConsulting/SSW.Website/assets/6656756/838d6b56-453a-40aa-a397-552e387b3050)

**Figure**: **editorial workflow for articles**
<br>
<img width="949" alt="image" src="https://github.com/user-attachments/assets/e95af34f-6539-4c5e-a3ae-29e70b8f9101">

**Figure** : **article with breadcrumbs enabled and Banner field populated**

igorgoldobin committed:
added new articles pages section

igorgoldobin committed:
Merge branch 'main' of https://github.com/SSWConsulting/SSW.Website into feature/new-articles-page

igorgoldobin committed:
merge

igorgoldobin committed:
list view styles

igorgoldobin committed:
Merge branch 'main' of https://github.com/SSWConsulting/SSW.Website into feature/new-articles-page

# Conflicts:
#	tina/tina-lock.json

wicksipedia committed:
update pnpm version

Calinator444 committed:
rebuilt pnpm lock

wicksipedia committed:
fix linting errors

wicksipedia committed:
fix warnings about using next/head in app router

wicksipedia committed:
use customLink instead of a tags to avoid full page loads

wicksipedia committed:
fixed issue with azure component not rendering at bottom of page

wicksipedia committed:
fix linting issue

wicksipedia commented:
I've cleaned up the linting errors, usages of `<img>`, and `<a>` tags, and the footer rendering.


This is using the pages router and not the app router as the PBI requests (more tech debt)

wicksipedia commented on pages/articles/index.tsx:
```diff
@@ -0,0 +1,127 @@
+          )}
+          {data.articlesIndex.showMicroservices && (
+            <div className="max-w-sm shrink pl-16">
+              <MicroservicesPanel
```
what is a microservices panel?

I think this is a loaded term and could have a better name

fenix2222 replied:
Renamed

wicksipedia commented on content/articles/test.mdx:
are we expecting to go live with this article?

fenix2222 replied:
Yes :)

wicksipedia commented on pages/articles/[filename].tsx:
```diff
@@ -0,0 +1,138 @@
+  });
+
+  return (
+    <RecaptchaContext.Provider
```
what do we need recaptcha for? I thought this was removed from the project?

fenix2222 replied:
removed

wicksipedia commented on components/microservices/microservicesPanel.tsx:
```diff
@@ -0,0 +1,36 @@
+              className="unstyled flex h-12 w-80 shrink-0 cursor-pointer items-center justify-center rounded bg-ssw-red px-5 uppercase text-white hover:opacity-70 max-sm:my-5 sm:w-fit"
+              href={url}
+            >
+              Learn more
```
`Learn more` as link text is not a good for SEO

https://www.seomechanic.com/optimize-cta-better-alternatives-to-learn-more/

fenix2222 replied:
it is now configurable

wicksipedia commented on content/articles/index/index.mdx:
```diff
@@ -0,0 +1,38 @@
+  title: 'SSW Articles - Sydney, Melbourne, Brisbane''s Premier Software Consultants'
+  description: "SSW has been developing software since 1990. Below are some of our great articles over the years. You can see much more specialized content in every area of software development on\_SSW Rules."
+title: SSW Articles
+showMicroservices: true
+microservicesTitle: 2-Day Pre-Migration Assessment Engagement
+microservicesDescription: >-
+  Get a solid foundation for your .NET 8 migration project, ensuring you are
+  well-prepared to tackle the migration with confidence.
+microservicesUrl: /
```
I think that this should this be an object instead of a few props related by a naming convention

Maybe call it sidebar?

fenix2222 replied:
done

wicksipedia replied:
This still looks like multiple props instead of an object
Also still called microservices… was code not pushed?

wicksipedia commented on content/articles/index/index.mdx:
```diff
@@ -0,0 +1,38 @@
+  Get a solid foundation for your .NET 8 migration project, ensuring you are
+  well-prepared to tackle the migration with confidence.
+microservicesUrl: /
+articles:
+  - title: Example Article
+    body: >
+      In today’s rapidly evolving tech landscape, CTOs are constantly seeking
+      ways to modernize and future-proof their applications. .NET 8 is emerging
+      as a top choice for modern app development, and here’s why.
+    pageURL: /articles/test
+    userName: Test User
+    userPosition: Software Developer
+    userImage: >-
+      https://www.ssw.com.au/people/static/Tiago-Araujo-Profile-d0d1c45c36b6f24eac389e3acc6a70ed.jpg
+  - title: Another Article
+    body: >
+      In today’s rapidly evolving tech landscape, CTOs are constantly seeking
+      ways to modernize and future-proof their applications. .NET 8 is emerging
+      as a top choice for modern app development, and here’s why.
+    pageURL: /articles/test
+    userName: Test User
+    userPosition: Test
+    userImage: >-
+      https://www.ssw.com.au/people/static/Tiago-Araujo-Profile-d0d1c45c36b6f24eac389e3acc6a70ed.jpg
```
violates DRY - this should be sourced from the specific articles

fenix2222 replied:
Done, sourcing it from content pages now

wicksipedia replied:
Article content still present, not sure that this was done

wicksipedia commented on pages/articles/[filename].tsx:
this should be app router enabled and not pages router

wicksipedia replied:
As discussed this will be carried as tech debt. Please log a pbi

wicksipedia commented on pages/articles/index.tsx:
this should be app router enabled and not pages router

fenix2222 replied:
as discussed, will be done later

wicksipedia committed:
remove duplicate property

igorgoldobin committed:
feat: refactored as per PR comments

igorgoldobin committed:
Merge branch 'feature/new-articles-page' of https://github.com/SSWConsulting/SSW.Website into feature/new-articles-page

igorgoldobin committed:
Merge branch 'main' of https://github.com/SSWConsulting/SSW.Website into feature/new-articles-page

igorgoldobin committed:
minor formatting fixes

Calinator444 committed:
updated articles to use static prop initialization

Calinator444 committed:
Updated articles breadcrumbs to match Figma prototype

Calinator444 committed:
Updated banner image prop to match Figma prototype

Calinator444 committed:
Updated types in articleCard

Calinator444 committed:
Remove duplicate Azure footer from Article template

Calinator444 committed:
Added pagination to articles list

Calinator444 committed:
removed unused imports for articles index page

fenix2222 committed:
Merge branch 'main' into feature/new-articles-page

Calinator444 committed:
Cleaned up react query hooks for articles

Calinator444 committed:
Cleaned up linting errors

Calinator444 committed:
Merge branch 'feature/new-articles-page' of https://github.com/SSWConsulting/SSW.Website into feature/new-articles-page

Calinator444 committed:
Update article author card to use next js images

Calinator444 committed:
Fixed linting warnings

Marxoz commented on app/layout.tsx:
Looks good, but these changes should be aligned with page routing layout component as well. Make sure they are aligned.

Calinator444 replied:
Done - I updated the `theme-color` component and deleted the `msapplication-TileColor` tag because it wasn't being generated in the dom for `/products`. However I kept the `/safari-pinned-tab.svg` meta tag because I found it was still in the `/products` pages.

See commit

https://github.com/SSWConsulting/SSW.Website/pull/2803/commits/75990c21e2a74105a44f2c81cdd085325e7f86a1

Calinator444 committed:
synchronized settings between app router layout component and pages router layout component

wicksipedia commented on tina/collections/articles.tsx:
```diff
@@ -0,0 +1,317 @@
+        },
+        {
+          type: "string",
+          label: "Page URL",
```
How does this work? Shouldn’t the article title form part of the url?

Calinator444 replied:
I'm not seeing this prop used anywhere so going to remove it and see if everything still works as intended.

Calinator444 replied:
Done - I've removed the prop because it's not used

wicksipedia commented on tina/collections/articles.tsx:
```diff
@@ -0,0 +1,317 @@
+        },
+        {
+          type: "string",
+          label: "User Name",
```
Source this data from the presenter collection to be DRY or log as tech debt. 

Same for the other user fields

Calinator444 replied:
Done - I upated the branch to pull the author data from the list.

wicksipedia reviewed (approved):
This is conditional approval assuming the above comments are resolved. I happened to notice these things on my phone with no testing - essentially rubber stamped

Calinator444 committed:
Removed unused history card prop from the articles schema

Calinator444 committed:
Updated aricles to fetch authors from tina

Calinator444 committed:
- Fixed articles siebar on responsive layouts
- Removed "show sidebar on mobile" and "full width body" fields article template

Calinator444 committed:
Fixed responsive layout on for articles template

Calinator444 committed:
fixed title on template article sidebar

Calinator444 committed:
Removed confusing fields from article template

Calinator444 committed:
fixed tina editing not working for articles index page

Calinator444 committed:
Remove unused articles index schema

Calinator444 committed:
Fixed articles rendering blank due to prop check in Tina

Calinator444 committed:
removed page blocks from article template because they weren't part of the figma prototype

Calinator444 committed:
Fixed centre alignment for responsive layout of articles index page

Calinator444 committed:
Fixed topside padding for sidebar on index page responsive view

Calinator444 committed:
renamed presenter field for articles to author

Calinator444 committed:
Updated instructions for adding authors to the list for better accuracy

Calinator444 committed:
committing changes to Tina schema

Calinator444 committed:
Merge branch 'main' into feature/new-articles-page

Calinator444 committed:
Fixed liting errors and removed unused _body field in Tina

Calinator444 committed:
Fixed linting errors

Calinator444 committed:
Fixed linting errors... for real this time

Calinator444 committed:
replaced single quotes with double quotes

Calinator444 committed:
fixed linting errors regarding tailwind

Calinator444 committed:
Fixed linting errors

Calinator444 committed:
removed unused es lint
